### ISSUE TITLE
Results - Heading level fix

### ISSUE THREAD

**Issue by angelina-smith** [GSA/px-benefit-finder#1038]
### What's expected

On the 'Your Potential Benefits' page, the Results heading should be an H2.

### What happened

Currently, the Results heading is an H3. (Skipped headings from H1 to H3 are a failure of WCAG 1.3.1.)

### How to replicate

Chrome. 
![image](https://github.com/GSA/px-benefit-finder/assets/142342010/0b7f86cd-4476-4f71-92df-23ad6061876b)

`<h3 class="result-view-heading font-family-sans" id="">Results</h3>`
